types/services.d.ts

```diff
@@ -566,9 +536,39 @@ declare namespace Realm {
                  *
                  * @see https://docs.mongodb.com/manual/reference/change-events/
                  */
-                watch(
-                    options?: WatchOptions<T["_id"]>,
-                ): AsyncGenerator<ChangeEvent<T>>;
+                watch(options?: {}): AsyncGenerator<ChangeEvent<T>>;
+
+                /**
+                 * Creates an asynchronous change stream to monitor this collection for changes.
+                 *
+                 * By default, yields all change events for this collection. You may specify at most one of
```

Comment:
I'm not sure if it is better to make the documentation cover all overloads (as you are doing here) or be specific to the overload selected? It seems like tsserver only shows the one overload at a time, so maybe this is best when writing code, even if the overload-specific docs are better for things like tooltips/hover text. 🤷🏻‍♂️ Have you seen an idiom among other projects?